### PR TITLE
Added Stage in the pipeline to deploy External environment

### PR DESCRIPTION
**Pull request by ajaygargdev**
Added Stage in the pipeline to deploy the External environment.

After completing PR to merge branch DEV to master. It will need to approve in the External environment by Michael, Per, and Sanjay.

image

Now it will look like:

image

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -77,4 +77,25 @@ stages:
...
+- stage: deploy_external
+  dependsOn: Build
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+  jobs:
+  - template: azure-pipelines-release-template.yml
+    parameters:
+      environmentNameSuffix: external
+      environment: External
+      environmentAutoDelete: false
+      azureSubscription: 'KMD Momentum Internal (77a2aca7-3b91-463d-9165-c6d0bb3689f9)'
```
Comment (amitxagarwal) [inline azure-pipelines.yml:91]: This should be the subscription of the external environment.

Comment (ajaygargdev) [reply azure-pipelines.yml:91]: All parameters will get change according to need. this is just a basic for external and will change according to requirment.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -77,4 +77,25 @@ stages:
       autoSwapSlots: true
       clientId: $(McaClientId)
```
Comment (ExploreMyCode) [inline azure-pipelines.yml:78]: questions @ajaygargdev

1. We are going to have multiple McaClientId for each Kammune id? Which one are we referring to here?
2. Same thing with client secret.

Comment (ajaygargdev) [reply azure-pipelines.yml:78]: For now, we are taking it from here.
ClientId we will get from appsettings according to Kommune Id. And secrets will come from keyvault

File: azure-pipelines.yml [hunk 3]
```diff
@@ -79,4 +79,25 @@ stages:
...
+      instanceId: external
+      resourceGroupLocation: westeurope
+      diagnosticSeqServerUrl: $(ShareddevDiagnosticSeqAServerUrl)
+      diagnosticSeqApiKey: $(ShareddevDiagnosticSeqApiKey)
+      webAppServicePlanSku: S1
+      webAppConfigAlwaysOn: false
+      autoSwapSlots: true
+      clientId: $(McaClientId)
+      clientSecret: $(McaClientSecret)
+      isDbRequired: $(DbRequired)
```
Comment (amitxagarwal) [inline azure-pipelines.yml:103]: not passing iskeyvaultrequired?

Comment (ajaygargdev) [reply azure-pipelines.yml:103]: the default value of `iskeyvaultrequired` is false. So only required in case we want Keyvault

**Pull request merged**